src/outbox/outboxActions.js

```diff
@@ -27,40 +28,54 @@ export const deleteOutboxMessage = (localMessageId: number) => ({
   localMessageId,
 });
 
+export const messageSuccessfulSend = (localMessageId: number) => ({
+  type: MESSAGE_SEND_SUCCESSFUL,
+  localMessageId,
+});
+
 export const trySendMessages = () => (dispatch: Dispatch, getState: GetState) => {
   const state = getState();
   if (state.outbox.length > 0 && !state.app.outboxSending) {
     dispatch(toggleOutboxSending(true));
     const auth = getAuth(state);
-    state.outbox.forEach(item =>
-      sendMessageApi(
-        auth,
-        item.type,
-        isPrivateOrGroupNarrow(item.narrow) ? item.narrow[0].operand : item.display_recipient,
-        item.subject,
-        item.content,
-        item.timestamp,
-        state.app.eventQueueId,
-      ),
-    );
+    state.outbox.forEach(async item => {
+      try {
+        await sendMessageApi(
```

Comment:
I merged this, because I realized your error handling was kinda connected to that.
I would still encourage you to do another PR, but consider how you will handle retries of failed messages too.